lib/index.js

```diff
@@ -195,7 +195,30 @@ function generateIssuerSubject(ctx) {
     };
 }
 
-exports.createStorage = async (ctx) => {
+exports.createStorage = async (ctx, useFunctionAccessToken) => {
+
+    if (useFunctionAccessToken) {
+        if (!ctx.fusebit || !ctx.fusebit.functionAccessToken) {
+            throw new Error('Creating storage context using a function access token requires that the caller is also in posession of the function access token.')
+        }
+        const storageCtx = {
+            fusebit_storage_id: `boundary/${ctx.boundaryId}/function/${ctx.functionId}/${uuid.v4()}`,
+            fusebit_storage_audience: ctx.body.baseUrl,
+            fusebit_storage_account_id: ctx.body.accountId,
+            fusebit_storage_subscription_id: ctx.body.subscriptionId,
+        };
+        try {
+            await Superagent.get(`${exports.generateStorageUrl(storageCtx)}/*?count=1`)
+                .set('Authorization', `Bearer ${ctx.fusebit.functionAccessToken}`);
+        }
+        catch (e) {
+            throw new Error(`Creating storage context that uses a function access token requires that the caller itself is in posession of a function access token with permissions to ${generateStoragePath(storageCtx)} storage id.`);
+        }
+        return storageCtx;
+    }
+
+    // Legacy mode - create PKI, issuer, and client with permissions to storage
```

Comment:
We can rip it out when the last add-on that has a dependency on the legacy mode is gone, or we have a new add-on model, whichever comes first. Until then, we have to support existing add-ons somehow. I don't quite care if it is an extra property or createFunction2 given the limited time frame this will ensure SDK has to live.

Reply:
Removed